web/profiles/custom/os2loop/modules/os2loop_upvote/os2loop_upvote.module

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/**
+ * @file
+ * The module file for os2loop_upvote.
+ */
+
+use Drupal\Core\Cache\Cache;
+
+/**
+ * Add number of upvotes to flag.
+ */
+function os2loop_upvote_preprocess_flag(array &$variables) {
+  $comment = $variables['flaggable'];
+  $flag_service = Drupal::service('flag.count');
+  $flag_counts = $flag_service->getEntityFlagCounts($comment);
+  if (isset($flag_counts['os2loop_upvote_upvote_button'])) {
+    $variables['upvotes'] = intval($flag_counts['os2loop_upvote_upvote_button']);
+  }
+  // Clear cache, so the UI reflects changes in top comment.
+  Cache::invalidateTags($comment->getCacheTags());
+}
+
+/**
+ * Find the top comment.
+ *
+ * If there is a comment makred as "correct answer", this is it.
+ * If there is no "correct answer", the top comment will be the most upvoted.
+ */
+function os2loop_upvote_preprocess_field(array &$variables, $hook) {
+  $flag_service = Drupal::service('flag.count');
+  $field = ['os2loop_question_answers', 'os2loop_post_comments'];
+  if (in_array($variables['field_name'], $field)) {
+    foreach ($variables['comments'] as $comment) {
+      if (isset($comment['#comment'])) {
+        $flag_counts = $flag_service->getEntityFlagCounts($comment['#comment']);
+        if (isset($flag_counts['os2loop_upvote_correct_answer'])) {
+          $top_comment = $comment;
+        }
+        else {
+          if (isset($flag_counts['os2loop_upvote_upvote_button'])) {
+            if (!isset($top_comment)) {
+              $top_comment = $comment;
+            }
+            $top_comment_flag_counts = $flag_service->getEntityFlagCounts($top_comment['#comment']);
+            if (isset($top_comment_flag_counts['os2loop_upvote_upvote_button']) && intval($flag_counts['os2loop_upvote_upvote_button']) > intval($top_comment_flag_counts['os2loop_upvote_upvote_button'])) {
+              $top_comment = $comment;
+            }
+          }
+        }
+      }
+    }
+    $top_comment['#top'] = TRUE;
+    array_unshift($variables['comments'], $top_comment);
+  }
+}
+
+/**
+ * Add classes for styling.
+ */
+function os2loop_upvote_preprocess_comment(array &$variables) {
+  if (isset($variables['elements']['#top'])) {
+    // @todo when styling change these classes to fit.
+    $variables['attributes'] += ['class' => ['rounded border border-success p-2 mt-2']];
+    $variables['#top'] = TRUE;
+  }
```

Comment:
Maybe we should just add a `top-comment` class and keep all styling in the style sheet?